### DISCUSSION TITLE
Sorting Question

### DISCUSSION THREAD

**Discussion by Grydpype** [ransome1/sleek#344] [Q&A] [answered]
Dear all,

I would like to get a better understanding of Sleek's sorting mechanism.

I have sorted my tasks with due-date first, followed by priority, and then the rest which I do not use for sorting. But I have now noticed several times that a (C) priority task appears before a (B) priority task, or a task with no priority appears before a task with a priority.

I don't know if this is a bug or if there is something in the settings or the content of the task that could lead to this; any suggestion what could be the reason for this?

KInd regards,
Oliver.

**Comment by ransome1** [comment]
Hi Oliver.

Can you provide a sample of todos that provoke this behaviour?

Thanks
ransome

**Reply by Grydpype** [reply]
Sure:
for example it occurs with these two tasks:
2022-03-13 Call gardener Max Green due:2022-03-14 @foo
(B) 2022-02-28 go shopping due:2022-03-14 rec:+2w @foo

I would expect the second task to come before the first if due-date is first and priority the second criteria.

Don't know if this helps but while anonymizing my data I noticed that apparently it has something to do with the context. When I switched the original context (alphabetically the first context) to another one, the order was correct. In the above examples, @foo is again the alphabetically first context.

Cheers,
Oliver.

**Comment by Grydpype** [comment]
I've played around some more.
The tasks have different creation dates, and moving "creation date" as sorting criteria above "context" causes  a rearrangement into the right order (even though the context is the same in both tasks).
And moving "priority" down below "project" (even though neither task has any projects) also cause a rearrangement into the right order.

**Reply by ransome1** [reply]
That's odd. Let me take a look into in when I have a quiet minute ;) But I think it would make sense and be very helpful if you could open an issue and describe your findings again so we can add it to the backlog. Thanks for reporting this.

**Comment by ransome1** [comment, marked as answer]
@Grydpype during the currently ongoing refactoring, I rewrote this particular sorting. I did some quick testing and also checked your example, which is now sorted correctly.

Can I ask you to take a look at this pre-release, if this works for you? You can download it here: https://github.com/ransome1/sleek/releases/tag/v1.1.7-rc.2 

Please be aware that major parts of the app are being reviewed and optimised currently. That will mostly be of benefit to the user, but can also introduce some new bugs. If you find anything besides the sorting issue, please report it :)

**Reply by Grydpype** [reply]
That sounds as if you were able to reproduce the sorting issue? For starters it's good to know that it's not a user error :-)

I'll try out the new release this evening & give you feedback; thanks for the quick response. Should it still occur I'll open an issue (which I haven't gotten around to yet ...)

Cheers,
Oliver.

**Reply by Grydpype** [reply]
I installed 1.1.7 and am as yet unable to recreate the issue, so thanks for that. If I notice anything new I'll open an issue.

Cheers,
Oliver.

**Reply by ransome1** [reply]
Glad to hear!
